boards: Add support WizNet W5500 EVB Pico2

soburi opened the pull request:
Add initial support for W5500 EVB Pico2

---
~note:~

~https://github.com/zephyrproject-rtos/zephyr/pull/84834#issuecomment-2645731473~

~I am aware that the `tests/net/conn_mgr_monitor/net.conn_mgr.dad` test will fail if the `netif:eth` function support is explicitly stated in yaml, but I will submit it following the settings of `w5500_evb_pico`, which only differs in soc.~

~https://github.com/zephyrproject-rtos/zephyr/issues/85816~

soburi committed:
boards: Add support WizNet W5500 EVB Pico2

Add initial support for W5500 EVB Pico2

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,269 @@
+.. _w5500_evb_pico2:
+
+Wiznet W5500 Evaluation Pico
+############################
```
```suggestion
.. zephyr:board:: w5500_evb_pico2
```

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,269 @@
+- 1 Watchdog timer peripheral
+- Wiznet W5500 Ethernet MAC/PHY
+
+
+.. figure:: img/w5500_evb_pico2_side.png
+     :align: center
+     :alt: W5500 Evaluation Board
+
+     Wiznet W5500_EVB_PICO evaluation board (Image courtesy of Wiznet)
```
```suggestion
```

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,269 @@
+Pin Mapping
+===========
+
+The peripherals of the RP2040 SoC can be routed to various pins on the board.
```
RP2040? 🤔

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,269 @@
+Overview
+********
+
+W5500-EVB-Pico is a microcontroller evaluation board based on the Raspberry
```
Pico --> Pico2? Looks like you need to do a global pass just like with RP2040 vs RP2350

soburi replied:
Ah, this is still draft...
I haven't written the documentation yet.
I plan to update it based on rp2040.

kartben replied:
Sorry didn't notice it was in draft, my bad!

soburi replied:
Don't mind. Thank you for caring.

soburi replied:
I have made some general revisions and removed the draft. Please review it again.

nordicjm commented on boards/wiznet/w5500_evb_pico2/Kconfig.defconfig:
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2025 TOKITA Hiroshi
```
commit typo *support

soburi replied:
Oops, Thank you for your catch.

nordicjm commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,269 @@
+Supported Features
+==================
+
+The w5500_evb_pico2 board configuration supports the following
```
```suggestion
The ``w5500_evb_pico2`` board supports the following
```

nordicjm commented on boards/wiznet/w5500_evb_pico2/w5500_evb_pico2.dtsi:
```diff
@@ -0,0 +1,170 @@
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
```
inconsistent spacing in file

soburi replied:
I have made some general revisions and removed the draft. Please review it again.

soburi commented:
@raveious 

If you know or can try it, could you let me know whether the `tests/net/conn_mgr_monitor/net.conn_mgr.dad` test passes on the w5500_evb_pico device?

When I tried to check the operation of w5500_evb_pico2, I ran the test using the following command, but it failed. So, I would like to know the result on w5500_evb_pico.

```sh
west twister --device-testing --device-serial /dev/ttyACM0 -p w5500_evb_pico2/rp2350a/m33 -s tests/net/conn_mgr_monitor/net.conn_mgr.dad -xOPENOCD=/usr/local/bin/openocd
```

raveious commented:
> @raveious
> 
> If you know or can try it, could you let me know whether the `tests/net/conn_mgr_monitor/net.conn_mgr.dad` test passes on the w5500_evb_pico device?
> 
> When I tried to check the operation of w5500_evb_pico2, I ran the test using the following command, but it failed. So, I would like to know the result on w5500_evb_pico.
> 
> ```shell
> west twister --device-testing --device-serial /dev/ttyACM0 -p w5500_evb_pico2/rp2350a/m33 -s tests/net/conn_mgr_monitor/net.conn_mgr.dad -xOPENOCD=/usr/local/bin/openocd
> ```

The suite is filtered out for that target, so to "technically' runs successfully.

soburi commented:
> > @raveious
> > If you know or can try it, could you let me know whether the `tests/net/conn_mgr_monitor/net.conn_mgr.dad` test passes on the w5500_evb_pico device?
> > When I tried to check the operation of w5500_evb_pico2, I ran the test using the following command, but it failed. So, I would like to know the result on w5500_evb_pico.
> > ```shell
> > west twister --device-testing --device-serial /dev/ttyACM0 -p w5500_evb_pico2/rp2350a/m33 -s tests/net/conn_mgr_monitor/net.conn_mgr.dad -xOPENOCD=/usr/local/bin/openocd
> > ```
> 
> The suite is filtered out for the w5500_evb_pico target, so to "technically' runs successfully.

Thank you for the information. I understand the background. However, It is seems we need to consideration...

soburi commented:
@kartben @nordicjm @raveious 
I have made some general revisions and removed the draft. Please review it again.

raveious commented on boards/wiznet/w5500_evb_pico2/doc/img/w5500_evb_pico2_side.png:
I could be wrong, but I think the silk screen says "W5500 EVB PICO" on it.

soburi replied:
Oh, It's a terrible mistake...
Thank you for catching it.
I fetched the correct image and replaced it from https://docs.wiznet.io/Product/iEthernet/W5500/w5500-evb-pico2.

raveious commented on boards/wiznet/w5500_evb_pico2/support/openocd.cfg:
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 TOKITA Hiroshi
```
2025?

soburi replied:
Thanks for your pointing out.
Fixed it.

kartben commented on boards/wiznet/w5500_evb_pico2/board.yml:
```diff
@@ -0,0 +1,6 @@
+board:
+  name: w5500_evb_pico2
+  full_name: W5500 Evaluation Pico2
```
W5500-EVB-Pico2

Same mistake was made for the Pico-based board -- this should be the commercial name

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,146 @@
+.. zephyr:board:: w5500_evb_pico2
+
+Wiznet W5500 Evaluation Board Pico2
+###################################
+
```
```suggestion
```

kartben commented on boards/wiznet/w5500_evb_pico2/doc/index.rst:
```diff
@@ -0,0 +1,146 @@
+- 1 Watchdog timer peripheral
+- Wiznet W5500 Ethernet MAC/PHY
+
+
+.. figure:: img/w5500_evb_pico2.webp
+     :align: center
+     :alt: W5500 Evaluation Board
+
+     Wiznet W5500_EVB_PICO evaluation board (Image courtesy of Wiznet)
```
Zephyr board sphinx directive takes care of adding the image to the overview box 

```suggestion
```

soburi committed:
doc: boards: wiznet: Use product name for `full_name` in board.yml

Use `W5500-EVB_Pico` - the commercial name of this board - for
`full_name` field of `board.yml`.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

nordicjm commented on boards/wiznet/w5500_evb_pico2/w5500_evb_pico2-pinctrl.dtsi:
```diff
@@ -0,0 +1,57 @@
+		group1 {
+			pinmux = <UART0_TX_P0>;
+		};
+		group2 {
```
newline needed

soburi replied:
Fixed it.

nordicjm commented on boards/wiznet/w5500_evb_pico2/w5500_evb_pico2-pinctrl.dtsi:
```diff
@@ -0,0 +1,57 @@
+		group1 {
+			pinmux = <SPI0_CSN_P17>, <SPI0_SCK_P18>, <SPI0_TX_P19>;
+		};
+		group2 {
```
newline needed

soburi replied:
Fixed it.

nordicjm commented on boards/wiznet/w5500_evb_pico/board.yml:
```diff
@@ -1,6 +1,6 @@
 board:
   name: w5500_evb_pico
-  full_name: W5500 Evaluation Pico
+  full_name: W5500-EVB-Pico
```
commits should be squashed

nordicjm replied:
Did not notice it's different boards

soburi replied:
Thank you for confirmation. I got it. I'l leave it.

nordicjm commented on boards/wiznet/w5500_evb_pico2/w5500_evb_pico2_rp2350a_m33_defconfig:
```diff
@@ -0,0 +1,11 @@
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=150000000
```
should be set in the soc `Kconfig` file from the dts property (can be fixed in another PR)

soburi replied:
Moved it into Kconfig.defconfig.

raveious commented:
> @raveious Sorry, but could you check again?

Same test(s) as before? I'll take a look later tonight.

soburi commented:
> > @raveious Sorry, but could you check again?
> 
> Same test(s) as before? I'll take a look later tonight.

The tests that were failing before now work correctly with the fix in https://github.com/zephyrproject-rtos/zephyr/issues/85816.
Your Change Request was still open, so I'd like you to check if there are any problems.
Thank you.
